src/Shared/FileUtilities.cs

```diff
@@ -421,6 +421,72 @@ internal static string FixFilePath(string path)
         }
 
         /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// Use a native shared method to massage file path. If the file is adjusted,
+        /// that qualifies is as a path.
+        /// </summary>
+        internal static string MaybeAdjustFilePath(string value)
+        {
+            if (NativeMethodsShared.IsWindows || string.IsNullOrWhiteSpace(value) ||
+                value.StartsWith("$(") || value.StartsWith("@(") || value.StartsWith("\\\\"))
+            {
+                return value;
+            }
+
+            // For Unix-like systems, we may want to convert backslashes to slashes
+            string newValue = Regex.Replace(value, @"[\\/]+", "/");
```

Comment:
@JeremyKuhne It depends. :)  If you're building a small set of projects, I doubt anyone would notice or care except the people tracking performance.  If you're trying to build tens of thousands of projects, it matters a LOT -- one of the biggest reasons MSBuild has such scalability issues is that its dynamic work discovery makes it so that it "can't" (= can't without significant re-architecting) throw away any project configuration information or target outputs, because it doesn't know what might want to consume them later in the mind.  

It's got a pretty simple cache to disk of those two sets of things, so it'll just slow way way way down paging rather than crash with OOM, but once you hit that point, it spends so much time writing to disk, GC'ing, and reading from disk, that the build becomes effectively unusable.  And the more memory pressure there is, the faster it hits that point.  

(That said, in this case since the strings ought to just be left floating to be GC'd, I doubt it would contribute much if at all to overall memory consumption; seems more likely that there'd be a small degradation in speed due to extra time spent in Regex + extra time spent GC'ing.)